### ISSUE TITLE
Change Sets/ Session Support?

### ISSUE THREAD

**Issue by gschleusner** [sqlitebrowser/dbhub.io#165]
Is there any chance that Sessions and Change Sets would be added to enable client side diffing and much smaller uploads.  

https://sqlite.org/sessionintro.html

They certainly won't work for all kinds of databases but it would be really powerful for datasets that are designed to take advantage.

**Comment by justinclift** [comment]
Yeah, it's a good idea.  @MKleusberg actually wrote a bunch of diff-ing code (in Go) already, which seems to work ok.

We'd probably need to add the same code (but a C++ version) to include in DB4S too, so it knows how to send diffs to the server.
